shell/platform/linux/fl_renderer.cc

```diff
@@ -52,29 +94,56 @@ static gboolean fl_renderer_real_start(FlRenderer* self, GError** error) {
   if (!eglChooseConfig(priv->egl_display, attributes, &egl_config, 1,
                        &n_config)) {
     g_set_error(error, fl_renderer_error_quark(), FL_RENDERER_ERROR_FAILED,
-                "Failed to choose EGL config");
+                "Failed to choose EGL config: %s", get_egl_error());
     return FALSE;
   }
   if (n_config == 0) {
     g_set_error(error, fl_renderer_error_quark(), FL_RENDERER_ERROR_FAILED,
-                "Failed to find appropriate EGL config");
+                "Failed to find appropriate EGL config: %s", get_egl_error());
     return FALSE;
   }
   if (!eglBindAPI(EGL_OPENGL_ES_API)) {
     g_set_error(error, fl_renderer_error_quark(), FL_RENDERER_ERROR_FAILED,
-                "Failed to bind EGL OpenGL ES API");
+                "Failed to bind EGL OpenGL ES API: %s", get_egl_error());
     return FALSE;
   }
 
   priv->egl_surface = FL_RENDERER_GET_CLASS(self)->create_surface(
       self, priv->egl_display, egl_config);
+  if (priv->egl_surface == nullptr) {
+    g_set_error(error, fl_renderer_error_quark(), FL_RENDERER_ERROR_FAILED,
+                "Failed to create EGL surface: %s", get_egl_error());
+    return FALSE;
+  }
   EGLint context_attributes[] = {EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE};
   priv->egl_context = eglCreateContext(priv->egl_display, egl_config,
                                        EGL_NO_CONTEXT, context_attributes);
+  if (priv->egl_context == nullptr) {
+    g_set_error(error, fl_renderer_error_quark(), FL_RENDERER_ERROR_FAILED,
+                "Failed to create EGL context: %s", get_egl_error());
+    return FALSE;
+  }
   EGLint value;
   eglQueryContext(priv->egl_display, priv->egl_context,
                   EGL_CONTEXT_CLIENT_VERSION, &value);
 
+  const EGLint shared_context_attribs[] = {EGL_WIDTH, 1, EGL_HEIGHT, 1,
+                                           EGL_NONE};
+  priv->resource_surface = eglCreatePbufferSurface(
+      priv->egl_display, egl_config, shared_context_attribs);
+  if (priv->resource_surface == nullptr) {
+    g_set_error(error, fl_renderer_error_quark(), FL_RENDERER_ERROR_FAILED,
+                "Failed to create EGL resource surface: %s", get_egl_error());
+    return FALSE;
```

Comment:
Resource context failures don't need to be fatal; we can log the error and then proceed without the resource context.